Automattic/wp-calypso
# ExternalLink to packages (Step 1): Add ExternalLink to components package

Related to https://github.com/Automattic/wp-calypso/issues/88181

## Proposed Changes

We are migrating `ExternalLink` to `@automattic/components` because it is a dependency in the `features-list` that we are also migrating to `@automattic/calypso-products`.

The same principle applies to previous related migrations (see https://github.com/Automattic/wp-calypso/pull/79257):

> The entirety of the migration is a 4 step process that creates easily digestible ( and revertible ) PRs to work with. Unfortunately, this process also eliminates the git history from migrated files unless the 4 steps are condensed into a single one. I don't have a great answer to this, but the tradeoff, to me, is worth the added safety of more understandable, compact, iterative pull requests.

This is Step 1: 
- adding a new component to `@automattic/components` package
- adding Storybook

## Dev Notes

The `ExternalLinkWithTracking` component now accepts a `recordTracksEvent` callback, which is optional. Internally, it will either use the passed callback or the one from `@automattic/calypso-analytics`.

## Testing Instructions



- Code review
- Unit tests
- Ensure Storybook stories render correctly with no errors: `run yarn storybook:start`

## Pre-merge Checklist
